sid/contacts.py

```diff
@@ -26,23 +26,26 @@ def calculate_contacts(contact_models, contact_policies, states, params, date):
         contacts (numpy.ndarray): DataFrame with one column for each contact model.
 
     """
-    columns = []
-    for model_name, model in contact_models.items():
+    contacts = np.zeros((len(states), len(contact_models)), dtype=DTYPE_N_CONTACTS)
+    can_have_contacts = (~states["needs_icu"]) & (~states["dead"])
+    participating_contacts = states[can_have_contacts]
+
+    for i, (model_name, model) in enumerate(contact_models.items()):
         loc = model.get("loc", params.index)
         func = model["model"]
-        cont = func(states, params.loc[loc])
+        cont = func(participating_contacts, params.loc[loc])
```

Comment:
Something like this?

```suggestion
        model_specific_contacts = func(participating_contacts, params.loc[loc])
```